src/diffusers/utils/hub_utils.py

```diff
@@ -265,12 +268,17 @@ def move_cache(old_cache_dir: Optional[str] = None, new_cache_dir: Optional[str]
         )
 
 
-def _add_variant(weights_name: str, variant: Optional[str] = None) -> str:
+def _add_variant(weights_name: str, variant: Optional[str] = None, add_suffix_keyword: bool = False) -> str:
     if variant is not None:
         splits = weights_name.split(".")
         splits = splits[:-1] + [variant] + splits[-1:]
         weights_name = ".".join(splits)
-
+    if add_suffix_keyword:
+        weight_name_split = weights_name.split(".")
+        if len(weight_name_split) == 2:
+            weights_name = weight_name_split[0] + "{suffix}." + weight_name_split[1]
+        elif len(weight_name_split) == 3:
+            weights_name = weight_name_split[0] + "{suffix}" + "." + ".".join(weight_name_split[1:])
```

Comment:
In theory, `weight_name_split[1]` and `".".join(weight_name_split[1:])` produce the same result in the case of `len(weight_name_split) == 2`. So you can reduce the number of cases to:

```
if add_suffix_keyword:
    weight_name_split = weights_name.split(".")
    weights_name = weight_name_split[0] + "{suffix}" + "." + ".".join(weight_name_split[1:])
```

(not tested myself)